xgcm/grid.py

```diff
@@ -1255,16 +1255,16 @@ def check_neighbor(link, position):
             self.axes[axis]._facedim = facedim
             self.axes[axis]._connections = axis_links
 
-    def _assign_metrics(self, metrics):
-        """
-        metrics should look like
-           {('X', 'Y'): ['rAC']}
-        check to make sure everything is a valid dimension
-        """
-
-        self._metrics = {}
+    _metrics = {}
+    
+    @property
+    def metrics(self):
+        return self._metrics
 
-        for key, metric_vars in metrics.items():
+    @metrics.setter
+    def metrics(self, key, value):
```

Comment:
As you seem to have realized :) we can actually remove the `for` loop below since `key` and `value` are scalars or single values.

What we will want to do is use the `for` loop in the `__init__` method to loop over every element in the `metrics` dictionary passed to `__init__`, and assign appropriately `self.metrics[key] = value`

Reply:
Do you mean changing:
```python
class Grid
...
if metrics is not None:
    self._assign_metrics(metrics)
```

into:

```python
    self._metrics(metrics)
```

Reply:
Something like

``` python
def __init__(..., metrics=None):
	self._metrics = {}
	if metrics is not None:
		# loop over metrics.items() and assign to self._metrics
```

Reply:
I'm trying to understand whether the `for loop` previously under _assign_metrics should be under `__init__` or the `@metrics.setter`. If the `__init__` for loop already assigns `self._metrics[key] = value`, what would be the purpose of the `@metrics.setter` method?

Reply:
It should be under `__init__`.

> If the __init__ for loop already assigns self._metrics[key] = value

It should assign to `self.metrics`, *not* `self._metrics` (so it will use the `setter` method, and make appropriate checks)